app/controllers/report_controller/menus.rb

```diff
@@ -336,103 +340,17 @@ def edit_reports
-      :style  => 'background: #fff;
-                  padding: 2px 0 6px 2px;
-                  color:#4b4b4b;
-                  font-size:12px;
-                  font-weight:bold;'
-    }
-
-    @tree = []
-    @branch = []
-    @parent_node = {}
-    menus.each do |r|
-      r.each_slice(2) do |menu, section|
-        @parent_node = TreeNodeBuilder.generic_tree_node(
-          "p__#{menu}",
-          menu,
-          'folder.png',
-          "Group: #{menu}",
-          :style => 'cursor: default;
-                     color: #4b4b4b;
-                     display: block;
-                     font-size:1.1em bold;
-                     font-weight:bold;
-                     height:22px;
-                     line-height: 22px;
-                     text-decoration:none;
-                     text-indent: 8px;
-                     vertical-align: top;
-                     width: 205px;'
-        )
-        if !section.nil? && section.class != String
-          section.each do |s|
-            if s.class == Array
-              s.each do |rec|
-                @branch_node = []
-                if rec.class == String
-                  @menu_node = TreeNodeBuilder.generic_tree_node(
-                    "s__#{menu}:#{rec}",
-                    rec,
-                    'folder.png',
-                    "Menu: #{rec}",
-                    :style => 'cursor:default;' # No cursor pointer
-                  )
-                else
-                  rec.each do |r|
-                    temp = rep_kids_menutree(r)
-                    @branch_node.push(temp) unless temp.nil? || temp.empty?
-                  end
-                  @menu_node[:children] = @branch_node unless @branch_node.nil? || @menu_node.include?(@branch_node)
-                end
-                @branch.push(@menu_node) unless @menu_node.nil? || @branch.include?(@menu_node)
-              end
-            elsif s.class == String
-              temp = rep_kids_menutree(s)
-              @branch.push(temp) unless temp.nil? || temp.empty?
-            end
-          end
-        end
-        @parent_node[:children] = @branch unless @branch.nil? || @parent_node.include?(@branch)
-        @tree.push(@parent_node) unless @parent_node.nil?
-        @branch = []
-      end
-    end
-    base_node[:children] = @tree
-    menu_roles_tree = base_node unless base_node.nil? || base_node.empty?
-    menu_roles_tree
-  end
-
-  def rep_kids_menutree(rec)
-    rpt = MiqReport.find_by_name(rec.strip)
-    @tag_node = {}
-    unless rpt.nil?
-      @tag_node = TreeNodeBuilder.generic_tree_node(
-        "r__#{rpt.id}_#{rpt.name}",
-        rpt.name,
-        'report.png',
-        "Report: #{rpt.name}",
-        :style => 'padding-bottom: 2px; padding-left: 0px;' # No cursor pointer
-      )
-    end
-    @tag_node
+  # Convenience for TreeBuilderMenuRoles
+  #
+  def build_menu_roles_tree(rpt_menu = nil)
+    TreeBuilderMenuRoles.new(
+      :menu_roles_tree, # name
```

Comment:
That seems inconsistent with TreeBuilder where the name comes first and the type second. The comments say the name has `_tree` and type does not. 🤔 

https://github.com/ManageIQ/manageiq/blob/master/app/presenters/tree_builder.rb#L18-L27